databroker/experimental/server_ext.py

```diff
@@ -305,6 +306,7 @@ def post_metadata(self, metadata, structure_family, structure, specs):
         }
 
         key = str(uuid.uuid4())
+        created_date = datetime.today()
```

Comment:
Prompted by yesterday's conversation I think we should expand the scope of this PR to include:

* A "Revisions" collection
* Tracking creation time and update time --- perhaps, as you suggested, only creation time on Document and only update time on DocumentRevision
* Metadata updates

I think each aspect may inform the details of how we do the others, so it is sensible to do them as a single PR.

Reply:
> Typically the full datetime is recorded, not just the date. Any particular reason for choosing date?

I thought it would be a good idea to record both date and time. This would give us a more exact information about the creation or update of a particular entry. Keep in mind that I'm importing `datetime` from the `datetime` package and not `date` which can only generate a date with no time

Reply:
I see. I have never seen `today()` used. I wrongly assumed it gave the date, like `datetime.now().date()`. I see now that it gives a datetime, similar to `datetime.now()`. According to the documentation [`now()` is preferred](https://docs.python.org/3/library/datetime.html#datetime.datetime.now). It does not state why it is preferred; perhaps it's just that the name is clearer.

Also, we need to consider timezones. For this kind of application, I think using UTC time is best. See that same documentation link above for useful information on how to get that.

Reply:
`datetime.today()` was changed to `datetime.now()`. Also, use of UTC was added